diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3f6e1d02..2e5a568f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -297,8 +297,11 @@ else()
     endif()
     FetchContent_Declare(curl
                          URL                    https://github.com/curl/curl/releases/download/curl-8_10_1/curl-8.10.1.tar.xz
+                                                https://vsagcache.oss-rg-china-mainland.aliyuncs.com/curl/curl-8.10.1.tar.xz
                          URL_HASH               SHA256=73a4b0e99596a09fa5924a4fb7e4b995a85fda0d18a2c02ab9cf134bebce04ee # the file hash for curl-8.10.1.tar.xz
-                         USES_TERMINAL_DOWNLOAD TRUE)   # <---- This is needed only for Ninja to show download progress
+                         USES_TERMINAL_DOWNLOAD TRUE
+                         INACTIVITY_TIMEOUT 5
+                         TIMEOUT 60)   # <---- This is needed only for Ninja to show download progress
     FetchContent_MakeAvailable(curl)
 
     restore_variable(DESTINATION CMAKE_CXX_CLANG_TIDY BACKUP CMAKE_CXX_CLANG_TIDY_BKP)
